src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+        if (player->stateFlags1 & PLAYER_STATE1_8000000) {
+            currentBoots++;
+        }
+        if (player->transformation == PLAYER_FORM_GORON) {
+            REG(48) = 0xC8;
+        }
+    } else if (currentBoots == PLAYER_BOOTS_GIANT) {
+        REG(48) = 0xAA;
+    }
+
+    bootRegs = D_801BFE14[currentBoots];
+    REG(19) = bootRegs[0];
+    REG(30) = bootRegs[1];
+    REG(32) = bootRegs[2];
+    REG(34) = bootRegs[3];
+    REG(35) = bootRegs[4];
+    REG(36) = bootRegs[5];
+    REG(37) = bootRegs[6];
+    REG(38) = bootRegs[7];
+    REG(39) = bootRegs[8];
+    REG(43) = bootRegs[9];
+    R_RUN_SPEED_LIMIT = bootRegs[10];
+    REG(68) = bootRegs[11];
+    REG(69) = bootRegs[12];
+    IREG(66) = bootRegs[13];
+    IREG(67) = bootRegs[14];
+    IREG(68) = bootRegs[15];
+    IREG(69) = bootRegs[16];
+    MREG(95) = bootRegs[17];
+
+    if (globalCtx->roomCtx.currRoom.unk3 == 2) {
+        R_RUN_SPEED_LIMIT = 500;
+    }
+
+    if ((player->actor.id == ACTOR_PLAYER) && (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+        scale = 0.015f;
+    } else {
+        scale = 0.01f;
+    }
+
+    Actor_SetScale(&player->actor, scale);
+}
+
+s32 Player_InBlockingCsMode(GlobalContext* globalCtx, Player* player) {
+    return (player->stateFlags1 & (PLAYER_STATE1_20000000 | PLAYER_STATE1_200 | PLAYER_STATE1_80)) ||
+           player->csMode != 0 || globalCtx->sceneLoadFlag == 0x14 || globalCtx->unk_18B4A != 0 ||
+           (player->stateFlags1 & PLAYER_STATE1_1) || (player->stateFlags3 & PLAYER_STATE3_80) ||
+           globalCtx->actorCtx.unk268 != 0;
+}
+
+s32 Player_InCsMode(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return Player_InBlockingCsMode(globalCtx, player) || player->unk_AA5 == 5;
+}
+
+s32 func_80123420(Player* player) {
+    return player->stateFlags3 & PLAYER_STATE3_80000000;
+}
+
+s32 func_80123434(Player* player) {
+    return player->stateFlags1 & (PLAYER_STATE1_40000000 | PLAYER_STATE1_20000 | PLAYER_STATE1_10000);
+}
+
+// Unused
+s32 func_80123448(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags1 & PLAYER_STATE1_400000) &&
+           (player->transformation != PLAYER_FORM_HUMAN || (!func_80123434(player) && player->unk_730 == 0));
+}
+
+s32 func_801234B0(Player* player) {
+    return player->transformation == PLAYER_FORM_GORON || player->transformation == PLAYER_FORM_DEKU;
+}
+
+s32 func_801234D4(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags2 & PLAYER_STATE2_8) || player->actor.speedXZ != 0.0f ||
+           ((player->transformation != PLAYER_FORM_ZORA) && (player->stateFlags1 & PLAYER_STATE1_8000000)) ||
+           ((player->transformation == PLAYER_FORM_ZORA) && (player->stateFlags1 & PLAYER_STATE1_8000000) &&
+            (!(player->actor.bgCheckFlags & 1) || (player->currentBoots < PLAYER_BOOTS_ZORA_UNDERWATER)));
```

Comment:
I mean, obviously this is terrible. But the last condition's the worst of all: it's literally unintelligible from this information.